include/GL/glui.h

```diff
@@ -34,11 +34,23 @@
 #ifndef GLUI_GLUI_H
 #define GLUI_GLUI_H
 
-// Having stdlib here first fixes some 'exit() redefined' errors on MSVC.NET
-// that come from old GLUT headers.
-#include <cstdlib>
+//Note: GCC is not required for this feature. It's just an indicator. 
+#ifdef __GNUC__
+//GLUI_GLX uses glXGetCurrentDisplay to find GLUT's chosen X display.
+//It will #include GL/glx.h and X11/Xlib.h. This covers all desktops.
+#ifndef GLUI_GLX
+#define GLUI_GLX
+#endif
+#endif
+
+#if defined(GLUT_API_VERSION) //|| defined(?)
+
+  // This is for applications that do not want GLUT
+  // #included for them and it should be used if an
+  // alternative back-end is selected. 
+  typedef struct GLUquadric GLUquadric;
```

Comment:
Well, GLU and OpenGL are included by GLUT's headers. So all mention of OpenGL and GLU etc need to be out of the headers. 

I removed an instance of glTranslate3f for example. Eventually OpenGL will be obsolete, and internally there will be a need to shift to a more OpenGL-ES like workflow. 

I'm not very interested in 3-D UIs right now. But I think that where I'm going is toward reproducing something like Microsoft's resource editor's experience, except I want to approach it not like a UI designer but like a 3-D modelling application. So what I see is 3-D dialogues being designed like 3-D models using the same software internals as a 3-D modeler uses.

The final abstraction will accept arbitrary geometry. So a GLUI might be good for games or VR like UIs although I'm not thinking in terms of like some of the attempts to make 3-D desktops. I'm thinking more in terms of developing self-contained products and then having resources for translating its UI into different languages and layouts to meet different needs.

A GLUI will need someway to distinguish itself from other UI frameworks. Especially if it ends up being largely a wrapper around wxWidgets as I expect it will. Then the mission has to be defined as either a 3-D flavored UI package or justify itself as being more user-friendly or approachable than wxWidgets. I think it should be a bit of both but that just the second option isn't enough.